### PR TITLE
Translation invariant version of 1516 !=> 1489

### LINKED ISSUE #675: GREEDY: Refute 1516 ->1489
A proof may be found at https://leanprover.zulipchat.com/#narrow/channel/458659-Equational/topic/Outstanding.20equations.2C.20v1/near/477860423

The formalized proof may be placed in the `ManuallyProved` folder. `ManuallyProved.lean` should be updated afterwards and the appropriate conjectures from `Conjectures.lean` should be removed.

### PR DESCRIPTION
**Pull request by b-reinke**
see #675

Closes #675

### CODE REVIEW

**Comment by b-reinke** [comment]
At the moment the proof is complete, but I still have to clean up the file. Some of the lemmas are also useful for other translation invariant approaches / facts about the free group. Maybe they can be collected somewhere else?

**Comment by teorth** [comment]
There doesn't seem to be a strong push right now to refactor the code, so if you could update `ManuallyProved.lean` and `Conjectures.lean` we can merge to at least make progress on the dashboard, and if some of the code here ends up being useful later (or conversely if some other formalized proof code is useful to golf the one here) we can do that in a future PR.

**Comment by b-reinke** [comment]
Ok, I did these changes, but I also started refactoring

**Comment by teorth** [comment]
OK, if you are still actively working on the code then I won't merge yet until you are done with refactoring for the time being.

**Comment by b-reinke** [comment]
I have extracted some statement about general free groups, in particular, there is now a statement that nontrivial elements of free groups have infinite order (in particular, they are not equal to their inverse). That was a bigger file (but I guess still of general interest), so I put it in the ForMathlib folder. Smaller lemmas that are needed by this discussion are in the Mathlib folder.

The statements for fresh generators I haven't extracted yet, not sure where to put them and if they are of sufficient general interest.

I also adapted the construction so that it can work as a basis for 1516 !=> 255.

File: equational_theories/ForMathlib/GroupTheory/FreeGroup/ReducedWords.lean [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+theorem reduced_iff_eq_reduce : reduced L ↔ reduce L = L := by
+  constructor
+  · intro h
+    rw [← reduced_min h]
+    exact reduce.red
+  · intro h
+    unfold reduced
+    rw [List.chain'_iff_all_append_cons_cons]
+    intro x y l₁ l₂ eq1 eq2
+    obtain ⟨x1, x2⟩ := x
```
Comment (madvorak) [inline equational_theories/ForMathlib/GroupTheory/FreeGroup/ReducedWords.lean:135]: Can you use lower indices please? Also, you can combine this line with the `intro` above.

**Pull request merged**